compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsPureTypeUniqueVisitor.java

```diff
@@ -83,7 +83,7 @@ private boolean isAnydata(BType type) {
     }
 
     private boolean isPureType(BType type) {
-        return type.tag == TypeTags.ANYDATA || type.tag == TypeTags.READONLY || isAnydata(type);
+        return type.tag == TypeTags.ANYDATA || isAnydata(type);
```

Comment:
I think we should replace all calls to `isPureType` with `isAnydata` as it doens't do anything other than calling `isAnydata`

Reply:
Updated in 6cfc931120f36be6ef71cfda98de8882e18c14e4